Sources/MapboxNavigation/IdleTimerManager.swift

```diff
@@ -0,0 +1,43 @@
+import Foundation
+import UIKit
+
+/// Manages UIApplication.shared.isIdleTimerDisabled
+final class IdleTimerManager {
+    final class Token {
+        private let onCancel: () -> Void
+
+        fileprivate init(_ onCancel: @escaping () -> Void) {
+            self.onCancel = onCancel
+        }
+
+        deinit {
+            onCancel()
+        }
+    }
+
+    static let shared: IdleTimerManager = .init()
+    
+    private var idleTokenCount: Int = 0 {
+        didSet {
+            updateIdleTimer()
+        }
+    }
+
+    private init() {}
+
+    /// Disabled idle timer while there is at least one Token alive.
+    func issueIdleToken() -> Token {
+        dispatchPrecondition(condition: .onQueue(.main))
```

Comment:
Should this precondition be moved to `updateIdleTimer` so that we check token deallocation too?

Reply:
I've changed the implementation to be thread-safe.